### PR TITLE
Add State Management: Zustand

### PR DESCRIPTION
**Pull request by ptsimon**
- also added a sample usage in src/states

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -16,12 +16,14 @@
     "@types/react-router-dom": "^5.1.7",
     "dotenv": "^8.2.0",
     "framer-motion": "^3.2.0",
+    "i": "^0.3.6",
```
Comment (jayehernandez) [inline frontend/package.json:19]: Pola ano to hahahha

Comment (ptsimon) [reply frontend/package.json]: removed.

**Comment by jayehernandez** [comment]
LGTM. Please resolve merge conflict

**Review by jayehernandez — APPROVED** [review]

File: frontend/package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.1",
     "typescript": "^4.1.3",
-    "web-vitals": "^0.2.4"
+    "web-vitals": "^0.2.4",
+    "zustand": "^3.4.1"
   },
```
Comment (ian-tabs) [inline frontend/package.json:26]: _solid_

File: frontend/src/states/sample_store.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import create from 'zustand';
+
+const useStore = create((set) => ({
+  isClicked: true,
+  toggle: () => set((state) => ({ isClicked: !state.isClicked })),
+}));
+
+export default useStore;
```
Comment (ian-tabs) [inline frontend/src/states/sample_store.tsx:8]: lgtm+

File: frontend/src/states/sample_component.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+      {' '}
+      {isClicked ? 'on' : 'off' }
+      <button type="button" onClick={() => toggle}>
+        Click Me!
+      </button>
+    </div>
+  );
+};
+
+export default Toggle;
```
Comment (ian-tabs) [inline frontend/src/states/sample_component.tsx:18]: lgtm

**Review by ian-tabs — COMMENTED** [review]
`in `**summary**, _lgtm_

**Review by ian-tabs — APPROVED** [review]
lgtm

**Review by jacintojosh — APPROVED** [review]
LGTM

**Comment by ptsimon** [comment]
salamat mga lods.

**Pull request merged**